### PR TITLE
Use LoggerRule

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/jenkins-test-harness/pull/27.

@reviewbybees

### CODE REVIEW

File: test/pom.xml [hunk 1]
```diff
@@ -186,6 +186,13 @@ THE SOFTWARE.
       <version>4.0</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpclient</artifactId>
```
Comment (jglick) [inline test/pom.xml]: Adapts to https://github.com/jenkinsci/jenkins-test-harness/pull/22.

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java [hunk 2]
```diff
@@ -15,27 +15,27 @@
...
+import hidden.jth.org.apache.http.client.AuthCache;
+import hidden.jth.org.apache.http.client.CredentialsProvider;
+import hidden.jth.org.apache.http.client.methods.CloseableHttpResponse;
+import hidden.jth.org.apache.http.client.methods.HttpGet;
+import hidden.jth.org.apache.http.client.protocol.HttpClientContext;
+import hidden.jth.org.apache.http.impl.auth.BasicScheme;
+import hidden.jth.org.apache.http.impl.client.BasicAuthCache;
+import hidden.jth.org.apache.http.impl.client.BasicCredentialsProvider;
+import hidden.jth.org.apache.http.impl.client.CloseableHttpClient;
+import hidden.jth.org.apache.http.impl.client.HttpClientBuilder;
```
Comment (jglick) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: Also adapting to https://github.com/jenkinsci/jenkins-test-harness/pull/22.

**Comment by jglick** [comment]
Uploaded the dependency [here](https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/main/jenkins-test-harness/2.13-SNAPSHOT/jenkins-test-harness-2.13-20160708.001646-1.jar), not sure why the CI job is not finding it. One of the many mysteries of Maven repository resolution and mirrors I suppose. @stephenc any idea?

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java (hunk 2 above)
Comment (recena) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: Ugly 😄

Comment (andresrc) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: undesired side-effect, I hope there are not many cases.

**Comment by andresrc** [comment]
:bee:

**Comment by jtnord** [comment]
so this is not functionally equivalent code as the `LoggerRule` setups up and uses a RingBuffer by default.

Not a bug - just pointing this out, that there is more creeping in than looks like on first pass. Perhaps the Rule needs to have a flag to enable/disable the log capturing?

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java (hunk 2 above)
Comment (jglick) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: Probably it was not necessary for @oleg-nenashev to use `JenkinsRule.WebClient` here. I am not aware offhand of other cases where these kinds of classes are used.

**Comment by jglick** [comment]
> this is not functionally equivalent code

Are you referring to a specific case?

> Perhaps the Rule needs to have a flag to enable/disable the log capturing?

Why would you need to disable log capture? Harmless if you are not asking for the records AFAIK.

**Comment by jtnord** [comment]
> Are you referring to a specific case?

The capturing of all log records in a buffer.

> Why would you need to disable log capture? 

it doesn't come for free, only one of the tests actually was using it (which shows the norm is to not care). 

I'm not bugging this - just pointing this out as something that could cause some issues - the tests already started consuming way more memory than they should on 2.x line...

File: test/src/test/java/jenkins/security/ApiTokenPropertyTest.java (hunk 2 above)
Comment (jglick) [inline test/src/test/java/jenkins/security/ApiTokenPropertyTest.java]: #2441 fixes this.

**Comment by jglick** [comment]
> The capturing of all log records in a buffer.

An unbounded buffer? There were no such tests that I know of. Some use case might arise for using an unbounded buffer. If so, would be easy enough to support it.

> it doesn't come for free

Pretty close, I think—just drops records into a preallocated array.

**Comment by jtnord** [comment]
:bee: as non of my comments are blocking.

**Comment by jtnord** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
LGTM 🐝 Needs a JTH release

**Pull request merged**